ddapm_test_agent/agent.py

```diff
@@ -386,55 +386,59 @@ async def _tracestats_by_session(self, token: Optional[str]) -> List[v06StatsPay
         return stats
 
     async def _integration_requests_by_session(
-        self, token: Optional[str], include_sent_integrations: Optional[bool] = False
+        self,
+        token: Optional[str],
+        include_sent_integrations: Optional[bool] = False,
+        get_all_requests: Optional[bool] = False,
     ) -> List[Request]:
         """Get all requests with an associated tested Integration."""
         integration_requests: List[Request] = []
-        for req in self._requests_by_session(token):
-            # see if the request was to update with a newly tested integration
-            if req.match_info.handler == self.handle_put_tested_integrations:
-                if "integration" not in req:
-                    data = json.loads(await req.read())
-                    integration_name = data.get("integration_name", None)
-                    integration_version = data.get("integration_version", None)
-                    req["integration"] = Integration(
-                        integration_name=integration_name,
-                        integration_version=integration_version,
-                        dependency_name=data.get("dependency_name", integration_name),
-                    )
-                    req["tracer_version"] = data.get("tracer_version", None)
-                    req["tracer_language"] = data.get("tracer_language", None)
-                    integration_requests.append(req)
-                elif include_sent_integrations:
-                    integration_requests.append(req)
-            # check if integration data was provided in the trace request instead
-            elif (
-                "_dd_trace_env_variables" in req
-                and "DD_INTEGRATION" in req["_dd_trace_env_variables"]
-                and "DD_INTEGRATION_VERSION" in req["_dd_trace_env_variables"]
-            ):
-                integration_name = req["_dd_trace_env_variables"]["DD_INTEGRATION"]
-                integration_version = req["_dd_trace_env_variables"]["DD_INTEGRATION_VERSION"]
-
-                if "integration" not in req:
-                    req["integration"] = Integration(
-                        integration_name=integration_name,
-                        integration_version=integration_version,
-                        dependency_name=req["_dd_trace_env_variables"].get("DD_DEPENDENCY_NAME", integration_name),
-                    )
+        for req in self._requests:
```

Comment:
I think we can just simplify all of this to

```suggestion
        requests = self._requests if token is None else self._requests_by_session(token)
        for req in requests:
```
?